File: onboarding-java/src/main/java/com/hr/onboardingjava/auth/dto/SignUpResponse.java
package com.hr.onboardingjava.auth.dto;

import com.hr.onboardingjava.auth.entity.User;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter @Setter @Builder
public class SignUpResponse {
    private String username;
    private String nickname;

    public static SignUpResponse from(User user) {
        return SignUpResponse.builder()
                .username(user.getUsername())
                .nickname(user.getNickname())
                .build();
    }
}

File: onboarding-kt/src/main/kotlin/com/hr/onboardingkt/auth/dto/SignUpResponse.kt
package com.hr.onboardingkt.auth.dto

import com.hr.onboardingkt.auth.entity.Users

data class SignUpResponse(
    val username: String,
    val nickname: String,
){
    companion object{
        fun from(users: Users): SignUpResponse{
            return SignUpResponse(
                username = users.username,
                nickname = users.nickname
            )
        }
    }
}